From ed397f571cbfb68a4d74e0ed55ec7e542988f99f Mon Sep 17 00:00:00 2001
From: Daniel Watkins <daniel.watkins@canonical.com>
Date: Mon, 23 Feb 2015 09:22:50 +0000
Subject: [PATCH 6/8] Always close the password server connection, even on
 failure.

---
 cloudinit/sources/DataSourceCloudStack.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/cloudinit/sources/DataSourceCloudStack.py b/cloudinit/sources/DataSourceCloudStack.py
index 25014aa3ab9f..0d5a7a30f594 100644
--- a/cloudinit/sources/DataSourceCloudStack.py
+++ b/cloudinit/sources/DataSourceCloudStack.py
@@ -67,10 +67,12 @@ class CloudStackPasswordServerClient(object):
         # Because we're reading off the socket directly, we can't re-use the
         # connection.
         conn = http_client.HTTPConnection(self.virtual_router_address, 8080)
-        conn.request('GET', '', headers={'DomU_Request': domu_request})
-        conn.sock.settimeout(30)
-        output = conn.sock.recv(1024).decode('utf-8').strip()
-        conn.close()
+        try:
+            conn.request('GET', '', headers={'DomU_Request': domu_request})
+            conn.sock.settimeout(30)
+            output = conn.sock.recv(1024).decode('utf-8').strip()
+        finally:
+            conn.close()
         return output
 
     def get_password(self):
-- 
2.11.0

